Systems.org

```diff
@@ -1058,7 +1059,7 @@ Now you can initialize your system using the following command:
 
 #+begin_src sh
 
-  guix system -L ~/.dotfiles/.config/guix/systems init path/to/config.scm /mnt
+  guix time-machine --channels=~/dotfiles/guix/channels.scm -- system -L ~/.dotfiles/.config/guix/systems init path/to/config.scm /mnt
```

Comment:
`guix system` does not offer the `-C` / `--channels=channels.scm` option, but this can be solved using `guix time-machine`, this ensure the file is used (this is not even neccesary but probably 101% assures the channels are being used instead of just relying on what "should" have been at `~/.config/guix/channels.scm`

Reply:
Thanks so much for pointing this out, I hadn't seen this approach before!  Does this mean I can skip the initial `guix pull` I run and just use this to pull the channels temporarily while the system config is being applied?

Reply:
That's interesting... but I'd keep using `guix pull && hash guix` before running `guix system init`, my original approach was to take away the need of _having to have_ a `~/.config/guix/channels.scm`, this means you may take this off:

```sh
mkdir -p ~/.config/guix
cp dotfiles/guix/channels.scm ~/.config/guix
```

Now, you may want to have that copied anyways if you're installing packages via `guix install` and need extra channels to be indexed.